### PR TITLE
Add option to disable update checker and set update interval

### LINKED ISSUE #150: Make update checker more flexible
Currently we run the update checker task every 6 hours. We don't push updates much frequent to utilize these 6 hours.
I guess, we can either make it configurable or run every 12/24 hours?
I'm open for suggestions.

### PR DESCRIPTION
**Pull request by StealWonders**
## Overview
Adds:
- option to enable/disable update checker
- option to specify update interval 

Fixes #150

## Description
I've gone ahead and added an option to disable the update checker all-together.
The update interval can now be specified in the configuration file. 
I'm unsure how to best communicate to the end-user that the value has a time unit of hours, as the config doesn't come with comments. Perhaps this info could be added to a wiki page?
I have also grouped the update checker settings together and added a config migration.

## Submitter Checklist

- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] New public fields and methods are annotated with `@since TODO`.
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by NotMyFault — COMMENTED** [review]
Thanks for the PR!

If the interval is customizable, I'd recommend limiting it to the prior 6h as the lowest amount possible, given it interfaces with the CF API and a low interval results in rate limits.

What is the notification value for? When would you want to disable notifications but enable the update checker? Habituated, this specific design pattern implies silent updates in the background, which is not the case, hence I'd recommend removing it.

Respectively since annotations should be filed as TODO until the actual release takes place.

**Comment by StealWonders** [comment]
> What is the notification value for?

This option was previously added so I decided to keep it around. Though you are correct, it does not make any sense.

**Comment by StealWonders** [comment]
Updated branch with the changes required. The old config key for enabling/disabling notifications will be removed on config upgrade. If this isn't wanted I can remove it :-)

File: src/main/java/com/thevoxelbox/voxelsniper/VoxelSniperPlugin.java [hunk 1]
```diff
@@ -83,6 +85,8 @@ public void onEnable() {
         PaperLib.suggestPaper(this);
         currentVersionTitle = getDescription().getVersion().split("-")[0];
         currentVersion = Double.parseDouble(currentVersionTitle.replaceFirst("\\.", ""));
+        // Don't register task if update check is unwanted
+        if (!voxelSniperConfig.isUpdateCheckerEnabled()) return;
```
Comment (Aurelien30000) [inline src/main/java/com/thevoxelbox/voxelsniper/VoxelSniperPlugin.java]: We don't impose a specific format, but something like this with explicit brackets may be better to fit to FAVS "conventions".
```java
if (!voxelSniperConfig.isUpdateCheckerEnabled()) {
    return;
}
```

**Review by Aurelien30000 — APPROVED** [review]
Thanks for this contribution :D

**Review by Aurelien30000 — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
Thanks for taking care of the issue!

**Pull request merged**